### PR TITLE
Start converting vue to composition api.

### PR DESCRIPTION
**Pull request by mshima**
- move typescript.spec.json to src/main/javascript/spec/typescript.json to make vscode happy.
- implement a useStore for migration purposes.
- convert many components to composition api.

Related to https://github.com/jhipster/generator-jhipster/issues/21547.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { Component, Vue } from 'vue-property-decorator';
+import { defineComponent } from 'vue';
 
-@Component
-export default class JhiFooter extends Vue {}
+export default defineComponent({
+  name: 'JhiFooter',
```
Comment (DanielFran) [inline generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs:4]: Maybe rename the component to:
'<%=jhiPrefixCapitalized%>Footer'

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs (hunk 1 above)
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs:4]: If possible I prefer to don't rename the components at big PRs.
Should the folder and file be renamed too?

File: generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+import { useStore } from '@/store';
 import AccountService from '@/account/account.service';
 <%_ if (enableTranslation) { _%>
 import TranslationService from '@/locale/translation.service';
 <%_ } %>
 
-import {
-  initVueXStore,
 <%_ if (enableTranslation) { _%>
+import {
```
Comment (qmonmert) [inline generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs:14]: on 1 line : 
`import { initI18N } from '@/shared/config/config';`

**Review by qmonmert — APPROVED** [review]

File: generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs (hunk 1 above)
Comment (mshima) [reply generators/vue/templates/src/main/webapp/app/core/jhi-footer/jhi-footer.component.ts.ejs:4]:
```suggestion
  name: '<%=jhiPrefixCapitalized%>Footer',
```

File: generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
   initI18N,
```
Comment (mshima) [inline generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs:15]:
```suggestion
import { initI18N } from '@/shared/config/config'; 
```

File: generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-<%_ } %>
 } from '@/shared/config/config';
```
Comment (mshima) [inline generators/vue/templates/src/test/javascript/spec/app/account/account.service.spec.ts.ejs:16]:
```suggestion
```

**Review by DanielFran — APPROVED** [review]

**Review by qmonmert — APPROVED** [review]

**Pull request merged**